examples/gg20_sign_client.rs

```diff
@@ -0,0 +1,477 @@
+                "round3",
+                serde_json::to_string(&(alpha_enc, miu_enc)).unwrap(),
+                uuid.clone()
+            )
+            .is_ok());
+            j += 1;
+        }
+    }
+
+    let round3_ans_vec = poll_for_p2p(
+        &client,
+        party_num_int,
+        THRESHOLD + 1,
+        delay,
+        "round3",
+        uuid.clone(),
+    );
+    let mut alpha_vec = vec![];
+    let mut miu_vec = vec![];
+    for i in 0..THRESHOLD {
+        let (l_alpha_enc, l_miu_enc): (AEAD, AEAD) =
+            serde_json::from_str(&round3_ans_vec[i as usize]).unwrap();
+        let out = aes_decrypt(&enc_key[i as usize], l_alpha_enc);
+        let bn = BigInt::from(&out[..]);
+        alpha_vec.push(ECScalar::from(&bn));
+
+        let out = aes_decrypt(&enc_key[i as usize], l_miu_enc);
+        let bn = BigInt::from(&out[..]);
+        miu_vec.push(ECScalar::from(&bn));
+    }
+
+    let input_stage4 = SignStage4Input {
+        alpha_vec_s: alpha_vec.clone(),
+        beta_vec_s: beta_vec.clone(),
+        miu_vec_s: miu_vec.clone(),
+        ni_vec_s: ni_vec.clone(),
+        sign_keys_s: res_stage1.sign_keys.clone(),
+    };
+    let res_stage4 = sign_stage4(&input_stage4).expect("Sign Stage4 failed.");
+    //broadcast decommitment from stage1 and delta_i
+    assert!(broadcast(
+        &client,
+        party_num_int,
+        "round4",
+        serde_json::to_string(&(res_stage1.decom1.clone(), res_stage4.delta_i,)).unwrap(),
+        uuid.clone()
+    )
+    .is_ok());
+    let round4_ans_vec = poll_for_broadcasts(
+        &client,
+        party_num_int,
+        THRESHOLD + 1,
+        delay,
+        "round4",
+        uuid.clone(),
+    );
+    let mut delta_i_vec = vec![];
+    let mut decom1_vec = vec![];
+    let mut j = 0;
+    for i in 1..THRESHOLD + 2 {
+        if i == party_num_int {
+            delta_i_vec.push(res_stage4.delta_i.clone());
+            decom1_vec.push(res_stage1.decom1.clone());
+        } else {
+            let (decom_l, delta_l): (SignDecommitPhase1, FE) =
+                serde_json::from_str(&round4_ans_vec[j]).unwrap();
+            delta_i_vec.push(delta_l);
+            decom1_vec.push(decom_l);
+            j += 1;
+        }
+    }
+
+    let delta_inv_l = SignKeys::phase3_reconstruct_delta(&delta_i_vec);
+    let input_stage5 = SignStage5Input {
+        m_b_gamma_vec: m_b_gamma_rec_vec.clone(),
+        delta_inv: delta_inv_l.clone(),
+        decom_vec1: decom1_vec.clone(),
+        bc1_vec: bc1_vec.clone(),
+        index: (party_num_int - 1) as usize,
+        sign_keys: res_stage1.sign_keys.clone(),
+        s_ttag: signers_vec.len(),
+    };
+    let res_stage5 = sign_stage5(&input_stage5).expect("Sign Stage 5 failed.");
+    assert!(broadcast(
+        &client,
+        party_num_int,
+        "round5",
+        serde_json::to_string(&(
+            res_stage5.R_dash.clone(),
+            res_stage5.R.clone(),
+            keypair.h1_h2_N_tilde_l_s.clone(),
```

Comment:
This should be sent during keygen (each party should end keygen with a vector of h1_h2_N)

Reply:
thanks for pointing this out. Changed the keygen to save this.